apps/dot-cdn/src/app/app.component.ts

```diff
@@ -1,59 +1,184 @@
+    }
+
+    // TODO: Missing type
+    changePeriod(event): void {
+        this.dotCdnStore.dispatchLoading({
+            loadingState: LoadingState.LOADING,
+            loader: Loader.CHART
+        });
+        this.setData(event.value);
+    }
+    /**
+     * Purges the entire cache
+     *
+     * @memberof AppComponent
+     */
+    purgePullZone(): void {
+        this.dotCdnStore.dispatchLoading({
+            loadingState: LoadingState.LOADING,
+            loader: Loader.PURGE_PULL_ZONE
+        });
+        this.dotCdnService.purgeCache([], true).subscribe(() => {
+            this.dotCdnStore.dispatchLoading({
+                loadingState: LoadingState.LOADED,
+                loader: Loader.PURGE_PULL_ZONE
+            });
+        });
+    }
+    /**
+     * Purges all the URLs in the array
+     *
+     * @memberof AppComponent
+     */
+    purgeUrls(): void {
+        const urls = this.urlsString.split(',').map((url) => url.trim());
+        this.dotCdnStore.dispatchLoading({
+            loadingState: LoadingState.LOADING,
+            loader: Loader.PURGE_URLS
+        });
+        this.dotCdnService.purgeCache(urls).subscribe(() => {
+            this.resetPurgeUrlsForm();
+        });
+    }
+
+    private resetPurgeUrlsForm(): void {
+        this.dotCdnStore.dispatchLoading({
+            loadingState: LoadingState.LOADED,
+            loader: Loader.PURGE_URLS
+        });
+        this.urlsString = '';
+        this.purgeZoneForm.setValue({ purgeUrlsTextArea: '' });
     }
 
     private setOptions(): void {
         this.options = {
-            title: {
-                display: true,
-                text: 'My Title',
-                fontSize: 16
-            },
-            legend: {
-                position: 'bottom'
+            responsive: true,
+            hoverMode: 'index',
+            stacked: false,
+            scales: {
+                yAxes: [
+                    {
+                        type: 'linear',
+                        display: true,
+                        position: 'left',
+                        id: 'y-axis-1'
+                    },
+                    {
+                        type: 'linear',
+                        display: true,
+                        position: 'right',
+                        id: 'y-axis-2',
+                        gridLines: {
+                            drawOnChartArea: false
+                        }
+                    }
+                ]
             }
         };
     }
 
-    private setData(): void {
-        this.data = {
-            labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
-            datasets: [
+    private setData(period: string): void {
+        this.dotCdnStore.dispatchLoading({
+            loadingState: LoadingState.LOADING,
+            loader: Loader.CHART
+        });
+        this.dotCdnService.requestStats(period).subscribe(({ stats }: DotCDNStats) => {
```

Comment:
This services should now live in the component store, the whole idea of the component store is that this component is now dumb and just subscribe to the store for data.